inc/parser/handler.php

```diff
@@ -357,6 +357,94 @@ function quote($match, $state, $pos) {
+        $open_char = '';
+        while ($pos < $max) {
+            $sign = $options[$pos];
+            if ($pos < $max-1 && ($sign != ',' || $in_string)) {
+                if (!$in_string && ($sign == "'" || $sign == '"')) {
+                    $in_string = true;
+                    $open_char = $sign;
+                } else if ($in_string && $sign == $open_char) {
+                    $in_string = false;
+                } else {
+                    $collected .= $sign;
+                }
+            } else {
+                if ($sign != ',') {
+                    // Collect the last sign ($pos == $max-1)
+                    if (!$in_string && ($sign == "'" || $sign == '"')) {
+                        $in_string = true;
+                        $open_char = $sign;
+                    } else if ($in_string && $sign == $open_char) {
+                        $in_string = false;
+                    } else {
+                        $collected .= $sign;
+                    }
+                }
+                $equal_sign = strpos($collected, '=');
+                if ($equal_sign === false) {
+                    // Just a flag, no value given, assign true.
+                    $collected = trim($collected);
+                    $result[$collected] = true;
+                } else {
+                    // We need to check if thee qual sign is inside the string
+                    $quote_sign = strpos($collected, "'");
+                    $double_quote_sign = strpos($collected, '"');
+                    $quote_pos = false;
+                    if ($quote_sign !== false) {
+                        $quote_pos = $quote_sign;
+                    }
+                    if ($double_quote_sign !== false &&
+                        ($quote_pos === false || $double_quote_sign < $quote_pos)) {
+                        $quote_pos = $double_quote_sign;
+                    }
+                    if ($quote_pos && $equal_sign > $quote_pos) {
+                        // Just a flag, no value given, assign true.
+                        $collected = trim($collected);
+                        $result[$collected] = true;
+                    } else {
+                        // key-value-pair
+                        $key = substr($collected, 0, $equal_sign);
+                        $key = trim($key);
+                        $value = substr($collected, $equal_sign+1);
+                        $value = trim($value);
+                        if ($value[0] == '"') {
+                            $value = trim($value, '"');
+                        } else if ($value[0] == "'") {
+                            $value = trim($value, "'");
+                        }
+                        $result[$key] = $value;
+                    }
+                }
+                $collected = '';
+            }
+            $pos++;
+        }
+        
+        if (count($result) == 0) {
+            return null;
+        }
+        return $result;
+    }
```

Comment:
This seems quite a lot of code. I wonder if it could be simplified. Maybe by making some assumptions on the used syntax? I recently implemented something similar using a regular expression to split up the options: https://github.com/cosmocode/csv/blob/master/helper.php#L53-L75 (it's slighlty more complex because the filter syntax is a bit complex in my plugin) but I think the simply key-value syntax used here should be easy to parse by regexp.